### DISCUSSION TITLE
A Kakoune-style meow tutorial

### DISCUSSION THREAD

**Discussion by Linerre** [meow-edit/meow#116] [General]
I finished about 70% I think. It looks nicer in the raw format:
https://raw.githubusercontent.com/Linerre/meow/scratch-post/SCRATCH_POST.org

1. keypad and beacon sections are not added yet.
2. A few basic to intermediate commands are not there either: `meow-join`, `meow-mark-word` and so on.

I'm afraid I'll have a busy Fri and hardly get my hands on it until Sat night. So feel free to make suggestions or changes.

I found `meow-expand-block` was changed to `meow-to-block`; also `meow-cancel` was changed back to `meow-cancel-selection`.  Wondering how to track such changes in the future.

**Comment by DogLooksGood** [comment]
The raw format looks great, it is not neccessary to be an org file.

 Any suggestion for where we put it?

---

We will avoid the breaking change. Since our version is greater than 1.0.

**Reply by DogLooksGood** [reply]
We can use it as *info*.

**Reply by DogLooksGood** [reply]
Add these headers
```
#+TITLE: Meow
#+OPTIONS: \n:t
#+TEXINFO_DIR_CATEGORY: Emacs
#+TEXINFO_DIR_TITLE: Meow: (meow.info)
#+TEXINFO_DIR_DESC: Yet another modal editing on Emacs
```

In your configuration, require `ox-texinfo`, use `C-c C-e i t` to export .texi file.
Rename the org file to `meow.org`, then the exported file will be `meow.texi`.

The texi file will be automatically loaded when compile the package.
(the package compilation will be triggered when a texi file is updated)

Then `Meow` will be accessible in `M-x info`.

The problem is multiple whitespaces will be collapsed  into one, unless you wrap it with `example block`.

![image](https://user-images.githubusercontent.com/11796018/145455665-309ad132-bc00-42ee-a6fa-62568a943498.png)

Another problem is that our keybindings are not limited to a single layout, but texi files are not dynamic.

**Reply by Linerre** [reply]
I'll test this out. I have added a few lines at the very top saying that
- this tutorial assumes QWERTY layout, but any other layouts should also work. 
- use the actual meow commands with their corresponding keybindings
